### PR TITLE
Alertbox Conditional & EntityRow Update

### PR DESCRIPTION
**Pull request by ailZhou**
### Description

Incremental progress on State and Territory Alert Box & Initiative Empty, Incomplete, Complete tickets. 

Swap the alertbox conditional to save data.
Note: There's an issue with the text needing some formatting, currently still thinking about how to do it. 

https://github.com/Enterprise-CMCS/macpro-mdct-mfp/assets/127151429/4e9e6a6d-a0b1-4498-b5ee-6620f00057c4

**It's using not the core validation atm but will be replaced with core later**

Note: This will still require a final update once core is in, but I wanted to get the code and function in for now so that there won't be merge issues later

### Related ticket(s)

MDCT-2776
MDCT-2795

---
### Important updates

N/A

---
### Author checklist

- [ ] I have performed a self-review of my code
- [ ] I have added [thorough](https://bit.ly/3zPrxuZ) tests, if necessary
- [ ] I have updated relevant documentation, if necessary
---

_convert to a different template: [test → val](?expand=1&template=test-to-val-deployment.md)_ | _[val → prod](?expand=1&template=val-to-prod-deployment.md)_

### CODE REVIEW

File: services/ui-src/src/components/alerts/getWPAlertStatus.tsx [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+};
+
+//store function calls here
+const alertStatusFunctions: AnyObject = {
+  initiatives: checkInitiativesTopics,
+};
+
+export const getWPAlertStatus = (report: ReportShape, entityType: string) => {
+  const fieldData = report?.fieldData;
+  var fn = alertStatusFunctions[entityType];
```
Comment (gmrabian) [inline services/ui-src/src/components/alerts/getWPAlertStatus.tsx:30]: is `var` making a comeback?! I've seen at least two team members use it now

**Review by gmrabian — DISMISSED** [review]
verified functionality
code lgtm

File: services/ui-src/src/components/alerts/getWPAlertStatus.tsx (hunk 1 above)
Comment (ailZhou) [reply services/ui-src/src/components/alerts/getWPAlertStatus.tsx:30]: nah, it's called I have a lack of good javascript foundations.

**Review by ntsummers1 — APPROVED** [review]
This looks good to me. The entityInfo might catch in the API when checking completionStatus. Not totally sure tbh, but if it does its just something I'll make note of and can fix over there.

**Review by britt-mo — APPROVED** [review]

File: services/ui-src/src/components/fields/TextField.tsx [hunk 2]
```diff
@@ -12,9 +23,68 @@ export const TextField = ({
...
   ...props
 }: Props) => {
+  const defaultValue = "";
+  const [displayValue, setDisplayValue] = useState<string>(defaultValue);
+
+  // get form context and register field
+  const form = useFormContext();
+  const fieldIsRegistered = name in form.getValues();
+
+  useEffect(() => {
```
[the remark this answers is not in the document]
Comment (ailZhou) [reply services/ui-src/src/components/fields/TextField.tsx:29]: codeclimate, i am ignoring you

**Pull request merged**